config/routes.rb

```diff
@@ -1,6 +1,15 @@
 Rails.application.routes.draw do
-   root "users#index"
-   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
-   end  
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root "users#index"
+
+  # Defines the root path route ("/")
+  # root "articles#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create]
+  end
+
+  resources :posts, only: [:index, :show, :new, :create] do
+    resources :comments, only: [:create]
+    resources :likes, only: [:create]
+  end
```

Comment:
[OPTIONAL] You have duplicated the routes for `posts`. It would be better to use only one `resources` for posts.